src/roadmap/v1/upcoming.py

```diff
@@ -74,22 +74,46 @@ class UpcomingOutputDetails(BaseModel):
 class UpcomingOutput(BaseModel):
     name: str
     type: UpcomingType
+    package: str
     release: str
     date: Date
     details: UpcomingOutputDetails
 
 
-class WrappedUpcoming(BaseModel):
+class WrappedUpcomingOutput(BaseModel):
     meta: Meta
     data: list[UpcomingOutput]
 
 
+class WrappedUpcomingInput(BaseModel):
+    meta: Meta
+    data: list[UpcomingInput]
+
+
 @lru_cache
 def read_upcoming_file(file: Path) -> list[UpcomingInput]:
     return TypeAdapter(list[UpcomingInput]).validate_json(file.read_text())
 
 
-def get_upcoming_data(
+def get_upcoming_data_no_hosts(settings: t.Annotated[Settings, Depends(Settings.create)]) -> list[UpcomingInput]:
+    return read_upcoming_file(settings.upcoming_json_path)
+
+
+@router.get(
+    "",
+    summary="Upcoming changes, deprecations, additions, and enhancements",
+)
+async def get_upcoming(data: t.Annotated[t.Any, Depends(get_upcoming_data_no_hosts)]) -> WrappedUpcomingInput:
+    return {
+        "meta": {
+            "total": len(data),
+            "count": len(data),
+        },
+        "data": data,
+    }
+
+
+def get_upcoming_data_with_hosts(
```

Comment:
nit: Put the routes at the bottom of the file and supporting classes/functions at the top of the file.